### DISCUSSION TITLE
Help test the new android app

### DISCUSSION THREAD

**Discussion by marcelklehr** [floccusaddon/floccus#962] [General]
Hey there,

since v4.8.x it is finally possible to run **floccus on Android** (with a slightly adjusted UI).

The experience on Android is not yet finalized and possibly buggy, but it's stable enough for me to let a few brave volunteers play with it and give feedback.

GoogleDrive is not yet supported, sadly -- I'm waiting for a patch to an underlying library, but it shouldn't be that long.

### Download
[<img alt="Google Play" src="https://play.google.com/intl/en_gb/badges/static/images/badges/en_badge_web_generic.png" height="70">](https://play.google.com/store/apps/details?id=org.handmadeideas.floccus)

Alternatively, you can **download the apk** on the release page: https://github.com/floccusaddon/floccus/releases/tag/v4.8.6

### Don't forget
Before you let it loose on your bookmarks, remember the first rule of fight club:

- Take a backup.

Really. Be kind to yourself and do it.

### Feedback welcome
Constructive feedback and bug reports are welcome on this thread. Preferably use voting to show support for a comment.

Cheers :blue_heart:

**Comment by marcelklehr** [comment]
Currently broken: 

- Back button doesn't work (closes the app instead of going back in the hierarchy)
- Sync changes are displayed only after an app restart

**Comment by stoneubi** [comment]
Hi Marcel,

great that you created an android app. 

Bookmark sync works. Found one bug on my phone. Once the 3 dots are selected next to any bookmark the "Edit or Delete" option will be displayed. Using the return button on my phone dosen't close the menu and the Edit and Delete selection still remains on the screen and cannot be closed. 

Thanks 
Stoneubi

**Comment by stoneubi** [comment]
Plus the return button dosen't work at all on my phone = the app cannot be closed using the built in android retun button. 

Setup: Redmi 6, Android V11.

**Comment by lorddoumer** [comment]
Great news, thank you so much! As soon as WebDav works, I'll help testing (since I've switched to .xbel-file due better speed) :)

**Reply by marcelklehr** [reply]
v4.8.4 now supports syncing via WebDAV -- see link in the original post :)

**Comment by uschmelmer** [comment]
Thank you very much for the effort!
I Found very high data consumption.
Foreground Data is 418MB and Background 134MB from October 1. - 18.
Sync interval is 15min.
The app also crashes very often.

**Reply by marcelklehr** [reply]
> I Found very high data consumption.

Mmmh, not sure what the data consumption is about :/ With Nextcloud bookmarks it should only check the hash of your bookmarks once every 15mins if nothing has changed, which shouldn't be so data intensive.

> The app also crashes very often.

Any recurring actions that trigger a crash?

**Reply by uschmelmer** [reply]
Now tested with 4.8.4:
> Any recurring actions that trigger a crash?
> 

For example, open a Bookmark redirects me to the browser. Ans some seconds later (while still in browser) it says that floccus has crashed.
Another example, is while beeing in the "Add Bookmark" dialog (manually selected from within the App). After a few seconds idling in this dialog, it crashes.
 > Mmmh, not sure what the data consumption is about
> 
It seems that just a single refresh of the list within floccus (which also takes quite long) uses about 90MB of data.

**Comment by DonPicciotto** [comment]
Everything works so far (latest apk, just synced and clicked on some links) with Nextcloud bookmarks. Thanks a lot for your work!

**Comment by gothmog123** [comment]
Working great with webdav! Thanks for your efforts! Any plans for fdroid/playstore inclusion?

**Reply by marcelklehr** [reply]
Yay :hearts: I definitely want to publish the app at some point, it still needs some polishing, though.

**Comment by misterhsp** [comment]
An option for accepting self-signed certificates would still be appropriate.

...

**Reply by marcelklehr** [reply]
Will be fixed with the next release: #997

**Comment by marcelklehr** [comment]
The app is now available on google play (see above).

**Reply by DonPicciotto** [reply]
It would be great to have it on F-Droid too. Anyway thanks for your work!

**Reply by marcelklehr** [reply]
Yep, that's in the works, too. :)

**Reply by marcelklehr** [reply]
Meanwhile you can already get the app from the IzzyOnDroid F-Droid repo (see https://floccus.org/download )

**Comment by ashutoshsaboo** [comment]
Hi @marcelklehr , thanks for all the great work! The GDrive option seems to be disabled on mobile (which is what I use on desktop), just wanted to check if that is already under the works? Or would that take longer time, if that's blocked by something else?

**Reply by marcelklehr** [reply]
Implementing GDrive support takes a bit longer, because OAuth works differently on Android than with web extensions, but should be done before the end of the year :)

**Reply by marcelklehr** [reply]
GDrive support is implemented and has landed on IzzyOnDroid as far as I can see. The infamous Game shop (aka Google Play) is taking a bit longer..

**Comment by DonPicciotto** [comment]
Yep, already updated from there! Thanks again 😁

10 dic 2021, 17:27 da ***@***.***:

**Comment by ybb8859** [comment]
Latest v4 In version 8.7, bookmarks in WebDAV cannot be synchronized on the mobile Wiki Browser!

**Reply by marcelklehr** [reply]
Could you open an issue for that? This thread is about the standalone android app.

**Comment by abe-101** [comment]
Working for me with nextcloud.
Feature request: ability to save bookmarks at specific locations? Also the ability to move bookmarks location

**Reply by marcelklehr** [reply]
Yep, definitely on the todo list, those two :)

**Comment by aecra** [comment]
Can we fix with the status bar color?

![Screenshot_2022-03-01-15-30-42-061_org handmadeid](https://user-images.githubusercontent.com/34883050/156124641-26baad17-32c3-4881-8644-c9ba7f9ee5eb.png)

**Reply by marcelklehr** [reply]
How would a fix look like? The status bar in grey as well? I'm not the most knowledgable person about android.

**Reply by aecra** [reply]
Grey instead of black for status bar looks well.

**Comment by rndhaz** [comment]
Hi marcelklehr, I am just playing a bit with the Android version 4.16 installed from F-Droid. Using it to sync bookmarks via WebDAV and working fine for me so far. Thanks for your great work!

However I can't seem to find a sync interval setting in the Android UI. Is that still missing? Rest of the app looks pretty much complete. If that option is indeed missing, does it use a default at the moment?

Another "issue" I found is, that the app is connecting to a Google service even when the 'allow network usage' option is not set. The domain is `content-autofill.googleapis.com`. If you haven't added any explicit code for that it might come from some included libs/framework, I guess. Anyhow, for a privacy concerned app this should imho be avoided somehow.

**Reply by marcelklehr** [reply]
Hi @rndhaz 

Thank you for your feedback! :) Currently the interval for android is always set to 15min, could you open a new issue for this? :pray: 

> Another "issue" I found is, that the app is connecting to a Google service even when the 'allow network usage' option is not set. 

This likely stems from the web view used on your system and not from the app itself. If you would open a new issue for this as well, then we can track it and coordinate with f-droid, which have been trying to look into this as well.

:hearts:

**Comment by yo-honey** [comment]
![Screenshot_20220902-151212.jpg](https://user-images.githubusercontent.com/46459324/188112547-24f02f54-170d-404b-935e-73ebac9c573e.jpg)

No option to export account?

Using 4.16.0

**Comment by SchwarzerA** [comment]
Missing the opportunity to select between merge/upload/download.

**Reply by marcelklehr** [reply]
Could you open a feature request in the issues for this?
